crates/bevy_pbr/src/render/light.rs

```diff
@@ -1994,6 +1989,20 @@ pub fn queue_shadows<M: Material>(
                     continue;
                 }
 
+                let mesh_layers = mesh_instance.shared.render_layers.as_ref();
+
+                // Skip the entity if it's not in the camera render layers.
+                if let (Some(camera_layers), Some(layers)) = (camera_layers, mesh_layers) {
```

Comment:
Typically the way we do this is to call `unwrap_or_default` on each of the respective layers, since a non-specified layer counts as layer 0. See `check_point_light_mesh_visibility` for example.